packages/rocketchat-api/server/v1/misc.js

```diff
@@ -168,3 +168,26 @@ RocketChat.API.v1.addRoute('shield.svg', { authRequired: false }, {
 		};
 	}
 });
+
+RocketChat.API.v1.addRoute('directory', { authRequired: true }, {
+	get() {
+		const { offset, count } = this.getPaginationItems();
+		const { sort, query } = this.parseJsonQuery();
+
+		const { text, type } = query;
+		const sortDirection = sort && sort === 1 ? 'asc' : 'desc';
+
+		const result = Meteor.runAsUser(this.userId, () => Meteor.call('browseChannels', {
+			text,
+			type,
+			sort: sortDirection,
+			page: offset,
+			limit: count
+		}));
+
+		if (result) {
```

Comment:
I think as a good practice, the unexpected condition should be treated specially and not the successful one.

so I will ask you to change this here, so the `if` will trigger the failure.